pandas/core/arrays/_ranges.py

```diff
@@ -32,57 +32,79 @@ def generate_regular_range(
 
     Returns
     -------
-    ndarray[np.int64] representing nanosecond unix timestamps
+    (tuple): containing:
+
+        values : ndarray[np.int64] representing nanosecond unix timestamps
+        tz : the timezone of the range
     """
     if isinstance(freq, Tick):
-        stride = freq.nanos
-        if periods is None:
-            b = Timestamp(start).value
-            # cannot just use e = Timestamp(end) + 1 because arange breaks when
-            # stride is too large, see GH10887
-            e = b + (Timestamp(end).value - b) // stride * stride + stride // 2 + 1
-            # end.tz == start.tz by this point due to _generate implementation
-            tz = start.tz
-        elif start is not None:
-            b = Timestamp(start).value
-            e = _generate_range_overflow_safe(b, periods, stride, side="start")
-            tz = start.tz
-        elif end is not None:
-            e = Timestamp(end).value + stride
-            b = _generate_range_overflow_safe(e, periods, stride, side="end")
-            tz = end.tz
-        else:
-            raise ValueError(
-                "at least 'start' or 'end' should be specified "
-                "if a 'period' is given."
-            )
-
-        with np.errstate(over="raise"):
-            # If the range is sufficiently large, np.arange may overflow
-            #  and incorrectly return an empty array if not caught.
-            try:
-                values = np.arange(b, e, stride, dtype=np.int64)
-            except FloatingPointError:
-                xdr = [b]
-                while xdr[-1] != e:
-                    xdr.append(xdr[-1] + stride)
-                values = np.array(xdr[:-1], dtype=np.int64)
-
+        start_value = Timestamp(start).value if start is not None else None
+        end_value = Timestamp(end).value if end is not None else None
```

Comment:
same as with timedeltas, id rather this be done in the calling function (along with the `tz = start.tz if start is not None else end.tz` on L48)